File: Clases 2018/Clases-C#/ClasePractica11_2018 (update tabla Bugs - asignar bug)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/ProductoDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class ProductoDao
    {
        public List<Producto> getAll()
        {
            List<Producto> lst = new List<Producto>();
            Producto oProducto = null/* TODO Change to default(_) if this is not a reference type */;

            foreach (DataRow row in BDHelper.getBDHelper().ConsultaSQL("Select * from Productos").Rows)
                lst.Add(map(row));

            return lst;
        }

        private Producto map(DataRow row)
        {
            Producto oProducto = new Producto();
            oProducto.id_Producto = Convert.ToInt32(row["id_producto"].ToString());
            oProducto.nombre = row["nombre"].ToString();
            return oProducto;
        }
    }
}

File: Clases 2018/Clases-C#/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/DataLayer/DAOs/ProductoDao.vb
Public Class ProductoDao
    Public Function getAll() As List(Of Producto)
        Dim lst As New List(Of Producto)
        Dim oProducto As Producto = Nothing

        For Each row As DataRow In BDHelper.getDBHelper.ConsultaSQL("Select * from Productos").Rows
            lst.Add(map(row))
        Next

        Return lst
    End Function

    Private Function map(ByRef row As DataRow) As Producto
        Dim oProducto As New Producto
        oProducto.id_Producto = Convert.ToInt32(row.Item("id_producto").ToString)
        oProducto.nombre = row.Item("nombre").ToString
        Return oProducto
    End Function
End Class
